firebase-firestore/src/main/java/com/google/firebase/firestore/Filter.java

```diff
@@ -0,0 +1,146 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RestrictTo;
+import com.google.firebase.firestore.core.FieldFilter;
+
+/** @hide */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+public class Filter {
+  private final FieldPath field;
+  private final FieldFilter.Operator operator;
+  private final Object value;
+
+  private Filter(@NonNull FieldPath field, FieldFilter.Operator operator, Object value) {
+    this.field = field;
+    this.operator = operator;
+    this.value = value;
+  }
+
+  public FieldPath getField() {
+    return field;
+  }
+
+  public FieldFilter.Operator getOperator() {
+    return operator;
+  }
+
+  public Object getValue() {
+    return value;
+  }
```

Comment:
I don't think you would want these to be part of the public API. You should annotate them with hide/RestrictTo so that we don't forget when we make the Filter class public.